kafka-plugins-common/src/main/java/io/cdap/plugin/batch/source/KafkaBatchConfig.java

```diff
@@ -190,74 +192,85 @@ public Schema getMessageSchema() {
+          collector.addFailure("The offset field must be of type long.", null);
         }
         offsetFieldExists = true;
       } else {
         messageFields.add(field);
       }
     }
     if (messageFields.isEmpty()) {
-      throw new IllegalArgumentException(
-        "Schema must contain at least one other field besides the time and key fields.");
+      collector.addFailure(
+        "Schema must contain at least one other field besides the time and key fields.", null);
     }
     if (getKeyField() != null && !keyFieldExists) {
-      throw new IllegalArgumentException(String.format(
-        "keyField '%s' does not exist in the schema. Please add it to the schema.", keyField));
+      collector.addFailure(String.format(
+        "keyField '%s' does not exist in the schema. Please add it to the schema.", keyField), null);
     }
     if (getPartitionField() != null && !partitionFieldExists) {
-      throw new IllegalArgumentException(String.format(
-        "partitionField '%s' does not exist in the schema. Please add it to the schema.", partitionField));
+      collector.addFailure(String.format(
+        "partitionField '%s' does not exist in the schema. Please add it to the schema.", partitionField), null);
     }
     if (getOffsetField() != null && !offsetFieldExists) {
-      throw new IllegalArgumentException(String.format(
-        "offsetField '%s' does not exist in the schema. Please add it to the schema.", offsetField));
+      collector.addFailure(String.format(
+        "offsetField '%s' does not exist in the schema. Please add it to the schema.", offsetField), null);
     }
     return Schema.recordOf("kafka.message", messageFields);
   }
 
   /**
    * @return broker host to broker port mapping.
    */
-  public Map<String, Integer> getBrokerMap() {
-    return parseBrokerMap(kafkaBrokers);
+  public Map<String, Integer> getBrokerMap(FailureCollector collector) {
+    return parseBrokerMap(kafkaBrokers, collector);
   }
 
   /**
    * Parses a given Kafka broker string, which is in comma separate host:port format, into a Map of host to port.
    */
-  public static Map<String, Integer> parseBrokerMap(String kafkaBrokers) {
+  public static Map<String, Integer> parseBrokerMap(String kafkaBrokers,
+      @Nullable FailureCollector collector) {
     Map<String, Integer> brokerMap = new HashMap<>();
     for (KeyValue<String, String> hostAndPort : KeyValueListParser.DEFAULT.parse(kafkaBrokers)) {
       String host = hostAndPort.getKey();
       String portStr = hostAndPort.getValue();
       try {
         brokerMap.put(host, Integer.parseInt(portStr));
       } catch (NumberFormatException e) {
-        throw new IllegalArgumentException(String.format(
-          "Invalid port '%s' for host '%s'.", portStr, host));
+        if (collector != null) {
+          collector.addFailure(String.format(
+            "Invalid port '%s' for host '%s'.", portStr, host), null);
```

Comment:
`.withConfigElement("kafkaBrokers", host + ":" + port)`

this should highlight that specific element in the list.

Reply:
it's better to create this message once and use it in both scenarios. Less likely for somebody in the future to modify one message without the other.